flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/converters/SqlProcedureCallConverter.java

```diff
@@ -133,9 +133,17 @@ private List<RexNode> reduceOperands(SqlCall sqlCall, ConvertContext context) {
                         context.getSqlValidator().getTypeFactory());
         List<RexNode> rexNodes = new ArrayList<>();
         for (int i = 0; i < sqlCall.operandCount(); i++) {
-            RexNode rexNode = context.toRexNode(sqlCall.operand(i), inputRowType, null);
+            RexNode rexNode =
```

Comment:
At first I wanted to understand why this change was needed. When I reverted this part of the change, I found that `ProcedureITCase#testNamedArguments` could run, but `SqlNodeToCallOperationTest#testWithNamedArguments` did not. When I dug deeper into the code, I found that another bug was in TypeInferenceOperandChecker#insertImplicitCasts (Line 158).
`final LogicalType argumentType = toLogicalType(callBinding.getOperandType(i));` should be replaced into 
`final LogicalType argumentType =
                    toLogicalType(
                            SqlTypeUtil.deriveType(callBinding, callBinding.operands().get(i)));`
You can re-produce this bug by `ProcedureITCase#testNamedArguments`.
Maybe we can unify the behavior of unwrapping more in advance.